dev/AppNotifications/AppNotificationContentBuilder/AppNotificationContentBuilder.idl

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.Windows.AppNotifications
+{
+    [contract(AppNotificationsContract, 2)]
+    runtimeclass ButtonConfig
+    {
+        ButtonConfig(String content);
+
+        // Gets the text to display on the button.
+        String Content{ get; };
+
+        // Gets app-defined string of arguments that the app can later retrieve once it is activated when the user clicks the button.
+        String Arguments { get; };
+
+        // Gets or sets an identifier used in telemetry to identify your category of action.This should be something like "Delete",
+        // "Reply", or "Archive".In the upcoming toast telemetry dashboard in Dev Center, you will be able to view how frequently
+        // your actions are being clicked.
+        String HintActionId{ get; };
+        ButtonConfig SetHintActionId(String hintActionId);
+
+        // Gets or sets the ID of an existing AppNotificationTextBox in order to have this button display to the right of the input.
+        String TextBoxId{ get; };
+        ButtonConfig SetTextBoxId(String textBoxId);
+
+        // Adds a key(without value) to the activation arguments that will be returned when the toast notification or its buttons are clicked.
+        ButtonConfig AddArgument(String key);
+
```

Comment:
Do we have an example of when this would be useful? In what situation would a developer use this?